include/envoy/api/io_error.h

```diff
@@ -25,6 +25,14 @@ class IoError {
     InProgress,
     // Permission denied.
     Permission,
+    // Message too big to send.
+    MessageTooBig,
+    // Kernel interrupt.
+    Interrupt,
+    // Given argument is not valid.
+    InvalidArgument,
```

Comment:
It seems like this should just be a RELEASE_ASSERT somewhere? Should this ever actually happen?

Reply:
It shouldn't happen in production. I added this more for debugging information. It happened to me when I try to sendmsg with a nonlocal v4 address even with FREEBIND.

Reply:
Can we just release assert this somewhere to aid with debugging vs. having it be a named error? I would prefer that we don't encourage someone to handle this error.

/wait

Reply:
done